pkg/odo/cli/url/list.go

```diff
@@ -71,9 +79,19 @@ func (o *URLListOptions) Run() (err error) {
 			//create headers
 			fmt.Fprintln(tabWriterURL, "NAME", "\t", "URL", "\t", "PORT")
 
-			for _, u := range urls.Items {
-				fmt.Fprintln(tabWriterURL, u.Name, "\t", url.GetURLString(u.Spec.Protocol, u.Spec.Host), "\t", u.Spec.Port)
+			for _, i := range localUrls {
+				var present bool
+				for _, u := range urls.Items {
+					if i.Name == u.Name {
+						fmt.Fprintln(tabWriterURL, u.Name, "\t", url.GetURLString(u.Spec.Protocol, u.Spec.Host), "\t", u.Spec.Port)
+						present = true
+					}
+				}
+				if !present {
+					fmt.Fprintln(tabWriterURL, i.Name, "\t", "<not created on cluster>", "\t", i.Port)
```

Comment:
May be also put a note at the end saying use `odo push` to create the urls that are marked <not created on cluster> ?

Reply:
fixed as below: 

```
$ odo url list
Found the following URLs for component nodejs-cpkl in application nodejs-ex-wxzu:
NAME                 URL                                                                       PORT
nodejs-cpkl-8080     http://nodejs-cpkl-8080-nodejs-ex-wxzu-myproject.192.168.42.95.nip.io     8080
nodejs-cpkl-8081     <not created on cluster>                                                  8081

Use `odo push` to create url on cluster
```